aten/src/ATen/native/cuda/IndexKernel.cu

```diff
@@ -59,6 +103,19 @@ void gpu_index_kernel(TensorIterator& iter, IntArrayRef index_size, IntArrayRef
 
     f(out_data, in_data, offset);
   });
+
+  if (cudaPeekAtLastError() != cudaSuccess) {
```

Comment:
What makes sure that every `assert(0)` 'break' inside the async-kernel execution is trapped? Isn't the kernel launched for aync-execution and`cudaPeekAtLastError()` might miss internal `assert()` calls?